pkg/platform/local/platform.go

```diff
@@ -714,13 +648,26 @@ func (p *Platform) GetContainerNameByCreateFunctionOptions(createFunctionOptions
 }
 
 func (p *Platform) getContainerHTTPTriggerPort(container *dockerclient.Container) int {
-	ports := container.HostConfig.PortBindings["8080/tcp"]
-	if len(ports) == 0 {
+	functionHostPort := dockerclient.Port(fmt.Sprintf("%d/tcp", abstract.FunctionContainerHTTPPort))
+
+	portBindings := container.HostConfig.PortBindings[functionHostPort]
+	ports := container.NetworkSettings.Ports[functionHostPort]
+	if len(portBindings) == 0 && len(ports) == 0 {
 		return 0
 	}
 
-	httpPort, _ := strconv.Atoi(ports[0].HostPort)
+	port := ""
+	if portBindings[0].HostPort != "" {
+
+		// if user specified a port, it will get here
+		port = portBindings[0].HostPort
+	} else {
+
+		// if no port was specified
+		port = ports[0].HostPort
+	}
 
+	httpPort, _ := strconv.Atoi(port)
```

Comment:
why skip the error check?

Reply:
at this point you must have either port or port binding, and they are presented as numbers, always

Reply:
I get that they should, correctness wise, but type-check-wise it's not a strong guarantee. they are strings, and atoi can fail. If that happens, you don't want that swallowed and the port coming back as nil